src/braket/default_simulator/openqasm/braket_result_pragmas.py

```diff
@@ -0,0 +1,128 @@
+from antlr4 import CommonTokenStream, InputStream
+from braket.ir.jaqcd import (
+    Amplitude,
+    DensityMatrix,
+    Expectation,
+    Probability,
+    Sample,
+    StateVector,
+    Variance,
+)
+from openqasm3.parser import parse
+
+from .dist.BraketPragmasLexer import BraketPragmasLexer
+from .dist.BraketPragmasParser import BraketPragmasParser
+from .dist.BraketPragmasVisitor import BraketPragmasVisitor
+
+
+class BraketPragmaNodeVisitor(BraketPragmasVisitor):
```

Comment:
Not a big deal, but this looks like the _perfect_ place for `singledispatch`.

Reply:
Actually this is the signature automatically created and used by antlr, just following an autogenerated interface here